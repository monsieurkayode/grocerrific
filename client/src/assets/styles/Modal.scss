.modal {
  ::-webkit-scrollbar {
    display: none;
  }

  @keyframes dropdown {
    from { top: -50%; }
    to { top: 50%; }
  }

  position: fixed;
  background: rgba(0,0,0,0.5);
  width: 100%;
  height: 100%;
  top: 0;
  left : 0;
  z-index: 50;

  .modal__main {
    position: fixed;
    width: 40%;
    min-width: px-to-rem(400px);
    max-width: px-to-rem(600px);
    background: #FFF;
    border-radius: 5px;
    top: 50%;
    left : 50%;
    transform: translate(-50%, -50%);
    padding: 40px 50px;
    box-shadow: 0 15px 20px -6px $light-grey-1;
    animation: dropdown 0.6s;

    .heading {
      font-family: $font-family-roboto;
      margin-bottom: 10px;
      text-transform: capitalize;
    }

    .close {
      float: right;
      padding: 0 8px;
      border-radius: 50%;
      transition: all 0.2s linear;
      background: $light-grey-1;
      cursor: pointer;
      outline: none;
      user-select: none;

      &:hover {
        transform: scale(1.2);
      }
    }

    .notice {
      margin: 20px auto;
      color: tomato;
      font-family: $font-family-roboto;
    }
  }
}