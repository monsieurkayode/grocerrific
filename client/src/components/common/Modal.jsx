import React from 'react';
import { node, string, func } from 'prop-types';

const Modal = ({ children, title, closeModal }) => (
  <div className="modal">
    <div className="modal__main">
      <div className="heading">
        <h2>{ title }
          <span
            onKeyPress={() => {}}
            role="button"
            tabIndex="0"
            onClick={() => closeModal()}
            className="close"
          >
            &times;
          </span>
        </h2>
      </div>
      { children }
    </div>
  </div>
);

Modal.propTypes = {
  children: node.isRequired,
  title: string.isRequired,
  closeModal: func.isRequired
};

export default Modal;
