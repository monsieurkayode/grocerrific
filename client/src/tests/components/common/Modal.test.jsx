import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../../../components/common/Modal';

const setup = () => {
  const props = {
    title: 'grocery cart',
    closeModal: jest.fn(),
    children: <p>Hello</p>
  };

  const wrapper = shallow(<Modal {...props} />);

  return { wrapper, props };
};

describe('Modal Component', () => {
  const { wrapper, props } = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a wrapping div with modal class', () => {
    expect(wrapper.find('.modal').length).toBe(1);
  });

  it('should have a h2 element with text grocery cart', () => {
    const modalHeader = wrapper.find('.heading h2');
    expect(modalHeader.length).toBe(1);
    expect(modalHeader.text()).toContain('grocery cart');
  });

  it('should have a close icon for closing modal', () => {
    const closeIcon = wrapper.find('.heading h2 span');
    expect(closeIcon.length).toBe(1);
  });

  it('should call closeModal function when the x icon is clicked', () => {
    const closeIcon = wrapper.find('.heading h2 span');
    closeIcon.simulate('click');
    expect(props.closeModal).toHaveBeenCalled();
  });
});
