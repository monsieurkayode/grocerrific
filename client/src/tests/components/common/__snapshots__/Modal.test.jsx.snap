// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should render correctly 1`] = `
<div
  className="modal"
>
  <div
    className="modal__main"
  >
    <div
      className="heading"
    >
      <h2>
        grocery cart
        <span
          className="close"
          onClick={[Function]}
          role="button"
          tabIndex="0"
        >
          Ã—
        </span>
      </h2>
    </div>
    <p>
      Hello
    </p>
  </div>
</div>
`;
